tests/functional/flows/test_create_flow.py

```diff
@@ -0,0 +1,220 @@
+            existing_identities[principal] = identity
+
+    return identity_set
+
+
+def value_for_field_from_output(name, output):
+    """
+    Return the value for a specified field from the output of a command.
+    """
+    match = re.search(rf"^{name}:[^\S\n\r]+(?P<value>.*)$", output, flags=re.M)
+    assert match is not None
+    return match.group("value")
+
+
+def test_create_flow_text_output(run_line):
+    # Load the response mock and extract metadata
+    response = load_response("flows.create_flow")
+    definition = response.metadata["params"]["definition"]
+    input_schema = response.metadata["params"]["input_schema"]
+    keywords = response.metadata["params"]["keywords"]
+    title = response.metadata["params"]["title"]
+    subtitle = response.metadata["params"]["subtitle"]
+    description = response.metadata["params"]["description"]
+    flow_administrators = response.metadata["params"]["flow_administrators"]
+    flow_starters = response.metadata["params"]["flow_starters"]
+    flow_viewers = response.metadata["params"]["flow_viewers"]
+
+    # Configure the owner identity
+    owner_identity = get_identity(response.json["flow_owner"].split(":")[-1], "pete")
+    identities = {response.json["flow_owner"]: owner_identity}
+    print(identities)
+
+    # Configure the identities for other roles
+    flow_administrator_identities = assign_identities_for_principal_set(
+        principal_set=flow_administrators,
+        existing_identities=identities,
+    )
+    flow_starter_identities = assign_identities_for_principal_set(
+        principal_set=flow_starters,
+        existing_identities=identities,
+    )
+    flow_viewer_identities = assign_identities_for_principal_set(
+        principal_set=flow_viewers,
+        existing_identities=identities,
+    )
+    print(identities)
+
+    load_response(
+        RegisteredResponse(
+            service="auth",
+            path="/v2/api/identities",
+            json={
+                "identities": list(identities.values()),
+            },
+        )
+    )
+
+    # Construct the command line
+    arguments = [
+        f"'{title}'",
+        f"'{json.dumps(definition)}'",
+    ]
+    for flow_administrator in flow_administrators:
+        arguments.extend(("--administrator", f"'{flow_administrator}'"))
+    for flow_starter in flow_starters:
+        arguments.extend(("--starter", f"'{flow_starter}'"))
+    for flow_viewer in flow_viewers:
+        arguments.extend(("--viewer", f"'{flow_viewer}'"))
+    for keyword in keywords:
+        arguments.extend(("--keyword", f"'{keyword}'"))
+    if input_schema is not None:
+        arguments.extend(("--input-schema", f"'{json.dumps(input_schema)}'"))
+    if subtitle is not None:
+        arguments.extend(("--subtitle", f"'{subtitle}'"))
+    if description is not None:
+        arguments.extend(("--description", f"'{description}'"))
+
+    result = run_line(f"globus flows create {' '.join(arguments)}")
```

Comment:
I see a lot of careful quoting here, and I think this is a good case for a refactor to use the non-shlex-splitting API for a click CliRunner. I'm going to suggest a refactor here, but it's untested so please check before assuming it's good. :smile: 

```suggestion
    command = [
        "globus",
        "flows",
        "create",
        title,
        json.dumps(definition),
    ]
    for flow_administrator in flow_administrators:
        command.extend(("--administrator", {flow_administrator}))
    for flow_starter in flow_starters:
        command.extend(("--starter", flow_starter))
    for flow_viewer in flow_viewers:
        command.extend(("--viewer", flow_viewer))
    for keyword in keywords:
        command.extend(("--keyword", keyword))
    if input_schema is not None:
        command.extend(("--input-schema", json.dumps(input_schema)))
    if subtitle is not None:
        command.extend(("--subtitle", subtitle))
    if description is not None:
        command.extend(("--description", description))

    result = run_line(command)
```

Reply:
I didn't know this was possible—perfect 👍